pagopa/io-app
# feat: [IOAPPCOM-44,IOAPPCOM-46] Mixpanel events for third party message attachments

⚠️ This PR depends on #4320 ⚠️

## Short description
This PR adds support for mixpanel events listed [here](https://docs.google.com/spreadsheets/d/1bXNRcTDKU-5l2RqCcwSbT7Y1M9QQTWtS_f-3-AIqSYk/edit#gid=1055214768).

## List of changes proposed in this pull request
Tracking events listed [here (Premium Tab)](https://docs.google.com/spreadsheets/d/1bXNRcTDKU-5l2RqCcwSbT7Y1M9QQTWtS_f-3-AIqSYk/edit#gid=54670005) have been added (rows 10 to 19 included)

THIRD_PARTY_MESSAGE_ATTACHMENT_COUNT | attachmentCount: numero di allegati | Inviato quando si ottiene il messaggio di terze parti (dalla chiamata al backend di IO) con la lista degli allegati
-- | -- | --
THIRD_PARTY_MESSAGE_ATTACHMENT_UNAVAILABLE | messageId: id del messaggioserviceId: id del servizio che ha mandato il messaggio | Inviato quando il backend di IO risponde con codice 404 alla chiamata per ottenere la lista degli allegati del messaggio di terze parti
THIRD_PARTY_MESSAGE_ATTACHMENT_DOWNLOAD_FAILED | messageId: id del messaggioserviceId: id del servizio che ha mandato il messaggio | Inviato quando il backend di IO risponde con un codice di errore generico (non gestito da altri eventi mixpanel) alla chiamata per ottenere la lista degli allegati del messaggio di terze parti
THIRD_PARTY_MESSAGE_ATTACHMENT_BAD_FORMAT | messageId: id del messaggioserviceId: id del servizio che ha mandato il messaggio | Inviato quando il backend di IO risponde con un codice 415 alla chiamata per ottenere la lista degli allegati del messaggio di terze parti
THIRD_PARTY_MESSAGE_ATTACHMENT_CORRUPTED_FILE | messageId: id del messagioserviceId: id del servizio che ha mandato il messaggio | Inviato quando il componente in app che mostra l'anteprima del PDF (allegato al messaggio di terze parti) non riesce a visualizzare quest'ultimo
THIRD_PARTY_MESSAGE_ATTACHMENT_PREVIEW_SUCCESS |   | Inviato quando il componente in app che mostra l'anteprima del PDF (allegato al messaggio di terze parti) visualizza correttamente quest'ultimo
THIRD_PARTY_MESSAGE_ATTACHMENT_SHOW_PREVIEW |   | Inviato quando l'utente preme sul tasto "Show Preview" nella bottom sheet inferiore che si apre, in seguito alla selezione di un allegato dalla lista degli allegati di un messaggio di terze parti
THIRD_PARTY_MESSAGE_ATTACHMENT_CANCEL |   | Inviato quando l'utente preme sul tasto "Cancel/Annulla" nella bottom sheet inferiore che si apre, in seguito alla selezione di un allegato dalla lista degli allegati di un messaggio di terze parti
THIRD_PARTY_MESSAGE_ATTACHMENT_DO_NOT_SHOW |   | Inviato quando l'utente attiva la spunta sulla checkbox "Non mostrare più" della bottom sheet inferiore che si apre, in seguito alla selezione di un allegato dalla lista degli allegati di un messaggio di terze parti
THIRD_PARTY_MESSAGE_ATTACHMENT_USER_ACTION | userAction: azione fatta dall'utente (una tra "open" "download" o "share") | Inviato quando un utente preme su uno dei tre (su android - uno su iOS) pulsanti che sono mostrati al di sotto dell'anteprima del PDF (allegato di un messaggio di terze parti)



## How to test
You can either use the dev-server (`withRemoteAttachments` setting in the `config`) or use the test credentials to try it out with the `io-backend` (contact me or the red unit to retrieve them).

Check that every of the above events is triggered and properly registered on mixpanel
